tests/integration/operator/profile_test.go

```diff
@@ -0,0 +1,142 @@
+package operator
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"testing"
+
+	"github.com/ghodss/yaml"
+
+	"istio.io/istio/operator/pkg/object"
+	"istio.io/istio/operator/pkg/tpath"
+	"istio.io/istio/operator/pkg/util"
+	"istio.io/istio/pkg/test/env"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/istioctl"
+	"istio.io/istio/pkg/test/framework/image"
+	"istio.io/istio/pkg/test/framework/resource/environment"
+)
+
+var (
+	IOPTestDataPath = filepath.Join(env.IstioSrc, "operator/cmd/mesh/testdata/manifest-generate/input")
+)
+
+func TestController(t *testing.T) {
+	framework.
+		NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			istioCtl := istioctl.NewOrFail(ctx, ctx, istioctl.Config{})
+			workDir, err := ctx.CreateTmpDirectory("operator-controller-test")
+			if err != nil {
+				t.Fatal("failed to create test directory")
+			}
+			compareInClusterWithGenerated(t, ctx, istioCtl, workDir, path.Join(IOPTestDataPath, "all_on.yaml"))
```

Comment:
The user should have a way to verify there install has completed, right? They certainly are not running these steps. Our test should reflect what the user will do as much as possible. I am not actually sure what they would do though. Check the status on the IOP? Run `verify-install`? 

I think its ok to have additional steps to verify more in depth since its a test, but we should at least run the user steps as well

Reply:
the thinking here is that since we're running pretty extensive e2e tests using istioctl, the bulk of the testing for operator can just compare that the install looks the same as istioctl. but yeah, in addition to this we need some e2e verification for at least some of the cases. verify-install would be good for all cases. maybe just one or two e2e tests out of the suite that runs against an istioctl install?

Reply:
I am fine with checking the istioctl install, my main comment is that users have the requirement "as a user of the operator, I need to understand when my install has completed or has issues". However the user does that, we should be doing that as well.

If we don't have a way to do that today, we should probably start tracking that and fill a TODO in here at least

Reply:
yep, I will add that part, verifying status is basic requirement, comparing incluster resources with manifest generate is more in depth

Reply:
I think what @ostromart meant in the above comment is that we should run some minimal e2e test with controller as installation